rust-k256/tests/signing.rs

```diff
@@ -0,0 +1,64 @@
+use k256::{
+    elliptic_curve::{point::AffineCoordinates, PrimeField},
+    FieldBytes, Scalar,
+};
+use plume_rustcrypto::{PlumeSignature, SecretKey};
+use rand_core::CryptoRng;
+use signature::RandomizedSigner;
+
+const message: &[u8; 29] = b"An example app message string";
```

Comment:
Can we add a note in docs or something that a protocol should set this message to be unique, and unique per nullifier? I.e. a protocol might put the name of the question being voted on here along with their protocol name for a ZK vote.

Reply:
That's a good one! What does message reuse brings? Is it only replay or would it let signing other messages without having the secret; or maybe anything else?

Reply:
No replay but you can deanonymize someones nullifier from the other place they posted and correlate that their identity must overlap with the other groups nullifier anonymity set.

Reply:
Why the point $r$ isn't included into the hash then? It would make each signature unique and move this foot-gun from those employing the protocol... :thinking:

Reply:
I totally trust you just trying to understand.

Deterministic as "to know `c` beforehand of signing"; or is it just the fact that V2 wouldn't work since it lacks the point $r$ value? Or something else?

I mean I kind of don't see yet the essential difference if signer adds to the message point $r$ instead of the info/data you mentioned. :thinking: